pkg/services/sqlstore/datasource.go

```diff
@@ -152,6 +162,14 @@ func UpdateDataSource(cmd *models.UpdateDataSourceCommand) error {
 			cmd.JsonData = simplejson.New()
 		}
 
+		if cmd.Uid == "" {
+			uid, err := generateNewDatasourceUid(sess, cmd.OrgId)
```

Comment:
If updating and uid not provided need to get the existing uid from db to be backward compatible, otherwise risk of uid keep changing.

Reply:
So it should be enough to actually not handle this right? So if Uid is empty during the update it won't be updated so will stay the same, if it is not empty will be overwritten.

Reply:
I think that's how xorm handles it, but please verify it works that way.

Reply:
Added a unit test for that.